first_task_search_engine/query_processing.py

```diff
@@ -0,0 +1,122 @@
+    query = query.lower().split(" ")
+    query = [lemm.lemmatize(w) for w in query if w != ""]
+    query_words = deque(query[::2])
+    query_operations = deque(query[1::2])
+    query_operations_set = set(np.unique(query_operations))
+    if query_operations_set.intersection(operations_set) != query_operations_set:
+        print("The query don't match the format.")
+        return {}
+
+    multiple_and_postings = []  # type:list
+    empty_postings_in_multi_end = False  # type:bool
+    if query:
+        word_left = query_words.popleft()  # type:str #first word in query
+        postings_left = get_postings(word_left)
+        while True:
+            try:
+                oper = query_operations.popleft()  # type:str
+                word_right = query_words.popleft()  # type:str
+                postings_right = get_postings(word_right)
+                if oper in ["or", "|"]:
+                    if multiple_and_postings:
+                        postings_left = intersect_many_postings_lists(
+                            multiple_and_postings
+                        )
+                        multiple_and_postings = []
+                    if postings_right:
+                        postings_left = union_postings_lists(
+                            postings_left, postings_right
+                        )
+                elif oper in ["not", "~"]:
+                    if multiple_and_postings:
+                        postings_left = intersect_many_postings_lists(
+                            postings_left, postings_right
+                        )
+                        multiple_and_postings = []
+                    if postings_right:
+                        postings_left = subtract_postings_lists(
+                            postings_left, postings_right
+                        )
+                else:
+                    if not empty_postings_in_multi_end:
+                        if multiple_and_postings:
+                            multiple_and_postings.append(postings_right)
+                        else:
+                            multiple_and_postings = [postings_left, postings_right]
+                    if not (postings_right or postings_left):
+                        empty_postings_in_multi_end = True
+                        multiple_and_postings = []
+            except IndexError:  # end of a query
+                if multiple_and_postings:
+                    postings_left = intersect_many_postings_lists(multiple_and_postings)
+                    multiple_and_postings = []
+                break
+    return postings_left
+
+
+def return_documents(query: str):
+    query = query.strip()
+    postings = get_postings_with_query(query)  # type:dict
+
+    if postings:
+        docs_dict = OrderedDict(
+            sorted(postings.items(), key=lambda x: x[1], reverse=True)
+        )  # sort documents by rank
+        docs_ids = np.array(list(docs_dict.keys()), dtype=int)[:5]
```

Comment:
```suggestion
        docs_ids = np.array(list(docs_dict.keys()), dtype=int)[:TOP_NUMBERS]
```
Magic numbers are not good solution :)

Reply:
Ok, that's a better idea